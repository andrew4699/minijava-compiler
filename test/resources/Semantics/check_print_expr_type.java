class a {

  public static void main(String[] args) {
    System.out.println(1);
  }
}

class b {
  public boolean valid() {
    int x; int[] y;
    y = new int[3];
    y[2] = 1;
    x = 1;
    System.out.println(x);
    System.out.println(x + 1 * 2 - 23);
    System.out.println(0);
    System.out.println(y[2]);
    System.out.println(1.0);
    System.out.println(1.0 * 2e20 - .5);

    return true;
  }

  public boolean invalid() {
    boolean x; int[] y;
    y = new int[2];
    x = false;
    System.out.println(false);
    System.out.println(x);
    System.out.println(y);

    System.out.println(11e55 * 1);
    System.out.println(1.0 + 1);
    System.out.println(1.0 - 1);

    return false;
  }
}
