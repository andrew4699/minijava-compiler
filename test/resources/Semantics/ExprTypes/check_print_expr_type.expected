[ExprTypesVisitor - 28]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 29]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 30]: Expression yielded type int[] where number was expected.
[ExprTypesVisitor - 32]: Expression yielded type double where int was expected.
[ExprTypesVisitor - 33]: Expression yielded type double where int was expected.
[ExprTypesVisitor - 34]: Expression yielded type double where int was expected.
